### PR TITLE
chore: support more query statement in extended query mode

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

- support other query statement('select','describe','show','with')
- support return describe error rather than panicking.
  In e2e test, some case will cause statement not create(such as giving a statement with semantics error). In this case if we panic when we can find the statement, the later test can't run. 
- modify some filed name of some message

This PR is prepare for introducing extended query mode e2e test. 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/frontend/src/session.rs [hunk 1]
```diff
@@ -615,10 +615,47 @@ impl Session for SessionImpl {
...
+                tracing::error!("failed to handle sql:\n{}:\n{}", sql, e);
+                e
+            })?,
+            Statement::ShowObjects(show_object) => match show_object {
+                ShowObject::Columns { table: _ } => {
+                    vec![
+                        PgFieldDescriptor::new("Name".to_owned(), TypeOid::Varchar),
+                        PgFieldDescriptor::new("Type".to_owned(), TypeOid::Varchar),
+                    ]
+                }
```
Comment (BowenXiao1999) [inline src/frontend/src/session.rs:629]: Do you take some ref on our previous code for this part?

File: src/frontend/src/session.rs [hunk 2]
```diff
@@ -615,10 +615,47 @@ impl Session for SessionImpl {
...
+                    vec![PgFieldDescriptor::new("Name".to_owned(), TypeOid::Varchar)]
+                }
+            },
+            Statement::ShowVariable { variable } => {
+                let name = &variable[0].value.to_lowercase();
+                if name.eq_ignore_ascii_case("ALL") {
+                    vec![
+                        PgFieldDescriptor::new("name".to_string(), TypeOid::Varchar),
+                        PgFieldDescriptor::new("setting".to_string(), TypeOid::Varchar),
+                        PgFieldDescriptor::new("description".to_string(), TypeOid::Varchar),
```
Comment (BowenXiao1999) [inline src/frontend/src/session.rs:640]:
```suggestion
                        PgFieldDescriptor::new("Name".to_string(), TypeOid::Varchar),
                        PgFieldDescriptor::new("Setting".to_string(), TypeOid::Varchar),
                        PgFieldDescriptor::new("Description".to_string(), TypeOid::Varchar),
```

File: src/frontend/src/session.rs (hunk 1 above)
Comment (ZENOTME) [reply src/frontend/src/session.rs:629]: Yes. This part I refer from the handle(frontend/src/handler/mod.rs/handle) and I extract the related part of PgFieldDescription.

File: src/frontend/src/session.rs (hunk 2 above)
Comment (ZENOTME) [reply src/frontend/src/session.rs:640]: I refer from https://github.com/singularity-data/risingwave/blob/2719006246c213f765d5ecc14596fa4979994de8/src/frontend/src/handler/variable.rs#L86.
If we want to unify, we need to unify in all place?

File: src/frontend/src/session.rs (hunk 1 above)
Comment (BowenXiao1999) [reply src/frontend/src/session.rs:629]: Add some comments for where refer from.

**Review by BowenXiao1999 — APPROVED** [review]

**Pull request merged**